flink/v2.1/flink/src/main/java/org/apache/iceberg/flink/sink/dynamic/DynamicWriteResultAggregator.java

```diff
@@ -160,12 +161,13 @@ byte[] writeToManifest(
     writeResults.forEach(w -> builder.add(w.writeResult()));
     WriteResult result = builder.build();
 
+    Table table = catalog.loadTable(TableIdentifier.parse(key.tableName()));
```

Comment:
We should cache, or make the info part of the Committable.

Committable seems like a better choice, as the serialized manifest file is different for the 2 versions